### PR TITLE
Fix miss serviceProvider ref in Encounter + tests

### PR DESCRIPTION
**Pull request by cragun47**
Signed-off-by: Brian Cragun <cragun@us.ibm.com>

Fixes the missing serviceProvider reference in Encounters.
Adjusts id's with spaces to have _ underscore for blanks.
Continues refactoring of touched tests to common getResource<type> bundle searches.

### CODE REVIEW

File: src/main/resources/hl7/resource/Organization_Simple.yml [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
-# (C) Copyright IBM Corp. 2020
+# (C) Copyright IBM Corp. 2020, 2021
 #
 # SPDX-License-Identifier: Apache-2.0
 #
 resourceType: Organization
+
+# The FHIR organization resource doesn't have any required fields.
+# However we do not want empty organization resources.
+# Thefore wherever organization is referenced make sure to verify the value is NOT_NULL.
```
Comment (LisaWellman) [inline src/main/resources/hl7/resource/Organization_Simple.yml:10]: I would reword the last part `the value is NOT_NULL.` to be either `the id or name is set`

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7EncounterFHIRConversionTest.java [hunk 2]
```diff
@@ -201,11 +201,92 @@ public void test_encounter_PV2_serviceProvider_idfix() {
...
         assertThat(organizations).hasSize(1);
 
-        Organization orgResource = getResourceOrganization(organizations.get(0));
+        Organization orgResource = ResourceUtils.getResourceOrganization(organizations.get(0),context);
         assertThat(orgResource.getId()).isEqualTo(providerString);
         assertThat(orgResource.getName()).isEqualTo("South Shore Hosptial Weymouth");
     }
 
+
+    // Test for serviceProvider reference in ADT_A01 message with only a PV1 segment
```
Comment (LisaWellman) [inline src/test/java/io/github/linuxforhealth/hl7/segments/Hl7EncounterFHIRConversionTest.java:210]: Suggest to beef up the comment to indicate that PV1.3.4.1 is used for both the Organization.id and Organization.name in this case.

**Review by LisaWellman — APPROVED** [review]
looks good!

File: src/main/resources/hl7/resource/Organization_Simple.yml (hunk 1 above)
Comment (cragun47) [reply src/main/resources/hl7/resource/Organization_Simple.yml:10]: done

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7EncounterFHIRConversionTest.java (hunk 2 above)
Comment (cragun47) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7EncounterFHIRConversionTest.java:210]: done
